### PR TITLE
Feature: admin tools

### PR DESCRIPTION
**Pull request by seangwright**
# Description

Adds UX to `/admin` path in client to view information about data persistence and clear/add bulk data records

Bulk data creation `Functions.Api` triggers also return stats about created data.

## Purpose

- [X] Feature
- [ ] Partial Feature
- [ ] Bugfix
- [ ] Docs, Config
- [ ] Chore / Format

## Change Type

- [ ] Major
- [X] Minor
- [ ] Fix

## Data Persistence Changes

n/a

## Configuration Changes

n/a

### CODE REVIEW

File: api/PayrollProcessor.Web.Api/Infrastructure/Responses/APIErrorResult.cs [hunk 1]
```diff
@@ -14,6 +13,8 @@ public APIErrorResult(string errorMessage = "") : base("")
                 : errorMessage;
 
             Value = new { Id = Guid.NewGuid(), Message = message };
+
+            StatusCode = 500;
```
Comment (seangwright) [inline api/PayrollProcessor.Web.Api/Infrastructure/Responses/APIErrorResult.cs:17]: This should have been set originally

File: client/src/app/shared/api-error.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { HttpErrorResponse } from '@angular/common/http';
+
+export function mapErrorResponse(
+  callback: (error: APIError) => void,
+): (response: HttpErrorResponse) => void {
+  return (response) => callback(response.error as APIError);
+}
+
+export interface APIError {
```
Comment (seangwright) [inline client/src/app/shared/api-error.ts:9]: This type matches up with the server side type

File: client/src/app/admin/admin.component.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        </div>
+      </form>
+      <ngx-spinner
+        name="resources"
+        bdColor="rgba(0, 0, 0, 0.8)"
+        size="medium"
+        color="#fff"
+        type="ball-spin"
+        [fullScreen]="false"
+      ></ngx-spinner>
```
Comment (KyleMcMaster) [inline client/src/app/admin/admin.component.html:67]: Nice! I have been looking to implement a spinner. Thanks for the example.

**Review by KyleMcMaster — APPROVED** [review]
LGTM

**Pull request merged**